### DISCUSSION TITLE
IPAM DNSsync not seem to be working

### DISCUSSION THREAD

**Discussion by Flabenelli** [sys4/netbox-plugin-dns#496] [Q&A] [answered]
I have netbox NetBox Community v4.1.7 with Netbox DNS  1.1.6.
Is running in the docker compose version.
I created zones, reverse zones and records, but the records don't seem to match with the IPs in the IPAM. 

I did rebuild_dnssync
reindex netbox_dns migrate ( after added the records, I forgot to doing by hand)
and setup_dnssync that added a custom field

DNSsync
Disable DNSsync 	X
TTL 	—
Disable PTR 	X

I'm doing something wrong?
Thanks

**Comment by peteeckel** [comment]
You didn't mention the most important item:

Did you create prefixes for your IP addresses and assign the prefixes you want records to be created for to the view(s) that your zones are assigned to?

This is all in the [documentation](https://github.com/peteeckel/netbox-plugin-dns/blob/main/docs/using_netbox_dns.md#ipam-dnssync).

**Comment by peteeckel** [comment]
By the way, if you set up your prefixes and views correctly there is no need to run anything apart from `setup_dnssync`.

**Comment by peteeckel** [comment]
And - don't add the address records manually, having that done automatically is the whole point of DNSsync :-)

**Comment by Flabenelli** [comment]
Hi, thanks for the quick reply. 

Yes I have prefixes configures and assigned to the default view. I prefer to not bother anyone so first I read the documentation thru out before asking questions here but I guess I misread something.

So What I did was: I uploaded all the records in the zone and then tried to add the prefix to the view but it complains that I have active records.

So your last comment put me in the right track. I deleted all records from the zone I previously uploaded and now I can add the prefix to the view and all the records I originally had, show up in the correct zone. Awesome!
I was expecting to work the other way around, you upload the records and if match the ip, loads the records to the IP.

So now I have a few questions: 

Now I have some of the records managed and many more in the Records tab
<img width="384" alt="Captura de pantalla 2024-12-13 a las 9 54 49" src="https://github.com/user-attachments/assets/029357ff-41d2-4377-9404-d56cf27d0adb" />

but if I try to add the record manually to the IP, it complains that the records already exist
<img width="741" alt="Captura de pantalla 2024-12-13 a las 9 56 16" src="https://github.com/user-attachments/assets/1ddca4a5-f2ba-4631-aca8-b4f910e952ea" />

So how can I sync the records? I need to delete the uploaded record in the zone, and then add it to the IP?

Thanks

**Comment by peteeckel** [comment]
Thanks, that explains the problem you're seeing. 

The issue with existing records exists both ways. If NetBox DNS doesn't permit creating duplicate records (which is the default) you can't enable DNSsync because the records it would create are already there, if it would (which you can allow by setting the config variable `enforce_unique_records` to `False`) you'll end up with duplicate records which you probably don't want either.

There is, however, [a third way](https://github.com/peteeckel/netbox-plugin-dns/blob/main/docs/using_netbox_dns.md#conflict-remediation). You can set `dnssync_conflict_deactivate` to `True`. The result is that as soon as DNSsync tries to create an A record that's already present it will be deactivated, which resolves the conflict and results in the creation of the managed record being allowed.

**Comment by Flabenelli** [comment]
Hmm, it seems that there is a nice feature that or it's missing, or I don't yet been able to get that is:

When you have "Managed Records", you have available the field "IPAM IP Address" that links the IP with the record. 
<img width="520" alt="Captura de pantalla 2024-12-13 a las 13 28 41" src="https://github.com/user-attachments/assets/ce10f6ee-aec0-407b-8b01-78cfa0e1dcb1" />


But if I upload a record and the IP in the IPAM doesn't have "DNS Name" , the record won't get managed and no "IPAM IP Address" exist. 
Is there a way yo get something like "IPAM IP Address" for no managed records? The only way I know right now would be, deleting all the records from the zone and updating all IP in the "DNS Name" one by one by hand.

To be clear, I'll be fine if the "DNS Name" field is empty as long as the link between record and ip exist.

Thanks!

**Reply by peteeckel** [reply, marked as answer]
It works the other way around :-)

You don't add a record manually. You link the prefix to the DNS view and fill in the "DNS Name" field for the IP address, and then the DNS record(s) will be created automatically as managed records.

**Reply by Flabenelli** [reply]
Ok, I guess there is a reason to not to sync both ways, but I find cumbersome to fill the "DNS name" field but really useful to bulk upload record using your plugin. But I guess it would be a feature request and not a question anymore. Thanks you for your help.

**Comment by peteeckel** [comment]
@Flabenelli All is not lost :-)

If you're not that interested in automatically generating the records for IP addresses, the DNSsync feature might not be the right solution for your use case. That feature was created because many users wanted to have address record created for them when they added IP addresses, not vice versa.

If you have a sensible way of mapping IP addresses to DNS records (which is surprisingly difficult to solve for the generic case), you might be better off with a custom script that does exactly that - look through all address records, find matching IP addresses in IPAM and then set "DNS name". There is an example script `DNS_IPAM_Updater.py` that contains code that does exactly that. You'll probably have to customise it a bit to exactly suit your needs, but it's a start.

Actually, you could even extend it to convert your existing DNS records to managed DNSsync records with a couple of tweaks. But that's really a special use case and probably won't make it into a feature - I'll think about it when I have the time, but probably it's too far from the generic use case.
